Plugin.cs

```diff
@@ -184,76 +184,97 @@ public void ReplacePlayer(Player playerToReplace)
-                Timing.CallDelayed(2f, () =>
+                float playerEnergy = scp.Energy;
+                int playerXp = scp.Experience;
+                int playerLevel = scp.Level;
+                Camera playerCamera = scp.Camera;
+                float playerBlackoutCoolDown = scp.BlackoutZoneCooldown;
+            
+                if (replacingPlayer.Role.Is(out Scp079Role newScp))
                 {
-                    if (spectator.Role.Is(out Scp079Role newScp))
-                    {
-                        newScp.Energy = playerEnergy;
-                        newScp.Experience = playerXP;
-                        newScp.Level = playerLevel;
-                    }
-                });
+                    newScp.Energy = playerEnergy;
+                    newScp.Experience = playerXp;
+                    newScp.Level = playerLevel;
+                    newScp.Camera = playerCamera;
+                    newScp.BlackoutZoneCooldown = playerBlackoutCoolDown;
+                }
             }
-            foreach (var item in playerToReplace.Items.ToArray())
+            
+            foreach (Item item in playerToReplace.Items.ToArray())
             {
-                CustomItem.TryGet(item, out CustomItem? custom);
-                spectator.GiveItemDelayed(item.Clone(), custom);
+                CustomItem.TryGet(item, out CustomItem? customItem);
+                replacingPlayer.GiveItemDelayed(item.Clone(), customItem);
                 item.Destroy();
             }
-            List<ItemType> ammo = playerToReplace.Ammo.Keys.ToList();
-            foreach (ItemType a in ammo)
-            {
-                Timing.CallDelayed(3f, () =>
-                {
-                    spectator.AddItem(a);
-                });
-            }
+        
+            foreach (KeyValuePair<ItemType, ushort> kvp in playerToReplace.Ammo)
+                Timing.CallDelayed(0.75f, () => replacingPlayer.SetAmmo(kvp.Key.GetAmmoType(), kvp.Value));
+            
             playerToReplace.ClearInventory();
-
-
-            spectator.Broadcast(10, $"<color=yellow>You have replaced an AFK player and become <color={playerToReplace.Role.Color.ToHex()}>{roleName}</color>.</color>");
-            Timing.CallDelayed(2f, () =>
-            {
-                spectator.ClearInventory(); // called before items are added
-                spectator.Teleport(position);
-                spectator.Health = playerHealth;
-                spectator.MaxHealth = playerMaxHealth;
-                spectator.ArtificialHealth = playerArtificialHealth;
-                spectator.MaxArtificialHealth = playerMaxArtificialHealth;
-                spectator.HumeShield = playerHumeShield;
-                spectator.Cuffer = playerCuffer;
-            });
-
+            playerToReplace.Vaporize();
+            
+            replacingPlayer.EnableEffect<SpawnProtected>(5f);
+            replacingPlayer.Teleport(playerToReplace.Position);
```

Comment:
Won't the player to replace not have a position anymore if they've been vaporized?

Reply:
You're right, I'll change it.